crates/router/src/core/payments/operations/payment_session_intent.rs

```diff
@@ -0,0 +1,372 @@
+            Some(id) => {
+                let customer = state
+                    .store
+                    .find_customer_by_global_id(
+                        &state.into(),
+                        id.get_string_repr(),
+                        &payment_data.payment_intent.merchant_id,
+                        merchant_key_store,
+                        storage_scheme,
+                    )
+                    .await?;
+                payment_data.email = customer.email.clone().map(common_utils::pii::Email::from);
+                Ok((Box::new(self), Some(customer)))
+            }
+            None => Ok((Box::new(self), None)),
+        }
+    }
+
+    #[instrument(skip_all)]
+    async fn make_pm_data<'a>(
+        &'a self,
+        _state: &'a SessionState,
+        _payment_data: &mut payments::PaymentIntentData<F>,
+        _storage_scheme: enums::MerchantStorageScheme,
+        _merchant_key_store: &domain::MerchantKeyStore,
+        _customer: &Option<domain::Customer>,
+        _business_profile: &domain::Profile,
+    ) -> RouterResult<(
+        PaymentsCreateIntentOperation<'a, F>,
+        Option<domain::PaymentMethodData>,
+        Option<String>,
+    )> {
+        Ok((Box::new(self), None, None))
+    }
+
+    async fn get_connector<'a>(
+        &'a self,
+        merchant_account: &domain::MerchantAccount,
+        state: &SessionState,
+        _request: &PaymentsSessionRequest,
+        payment_intent: &storage::PaymentIntent,
+        merchant_key_store: &domain::MerchantKeyStore,
+    ) -> CustomResult<api::ConnectorChoice, errors::ApiErrorResponse> {
+        let db = &state.store;
+        let all_connector_accounts = db
+            .find_merchant_connector_account_by_merchant_id_and_disabled_list(
+                &state.into(),
+                merchant_account.get_id(),
+                false,
+                merchant_key_store,
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Database error when querying for merchant connector accounts")?;
+        let profile_id = &payment_intent.profile_id;
+        let filtered_connector_accounts = helpers::filter_mca_based_on_profile_and_connector_type(
+            all_connector_accounts,
+            profile_id,
+            common_enums::ConnectorType::PaymentProcessor,
+        );
+        let mut connector_and_supporting_payment_method_type = Vec::new();
+        filtered_connector_accounts
+            .iter()
+            .for_each(|connector_account| {
+                let res = connector_account
+                    .payment_methods_enabled
+                    .clone()
+                    .unwrap_or_default()
+                    .into_iter()
+                    .map(|payment_methods_enabled| {
+                        payment_methods_enabled
+                            .parse_value::<PaymentMethodsEnabled>("payment_methods_enabled")
+                    })
```

Comment:
```suggestion
                let res = connector_account
                    .payment_methods_enabled
                    .clone()
                    .unwrap_or_default()
                    .into_iter()
                    .map(|payment_methods_enabled| {
                        payment_methods_enabled
                            .parse_value::<PaymentMethodsEnabled>("payment_methods_enabled")
                    })
```
use a impl based function to do this logic